src/request.m

```diff
@@ -295,8 +295,11 @@ - (void) _updateWindow: (NSInteger) delta withBlock: (void (^)()) block {
       window_out_queue_ = [NSMutableArray arrayWithCapacity: 16];
 
     // Retry on positive window_out
+    __weak __typeof(self) weakSelf = self;
```

Comment:
That's the thing. This message will happen sometime in a future, when the server will acknowledge previously sent data. It will unlikely coincide with the `refcount == 0`.